src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/IllegalTypeCheck.java

```diff
@@ -130,13 +130,88 @@
  * </li>
  * </ul>
  * <p>
- * To configure the check so that it ignores getInstance() methods:
+ * Default Configuration:
+ * </p>
+ * <pre>
+ * &lt;module name=&quot;IllegalType&quot;/&gt;
+ * </pre>
+ * <pre>
+ * public class Test extends TreeSet { // violation
+ *   public &lt;T extends java.util.HashSet&gt; void method() { // violation
+ *
+ *     LinkedHashMap&lt;Integer, String&gt; lhmap =
+ *     new LinkedHashMap&lt;Integer, String&gt;(); // violation
+ *     TreeMap&lt;Integer, String&gt; treemap =
+ *     new TreeMap&lt;Integer, String&gt;(); // violation
+ *     Test t; // OK
+ *     HashMap&lt;String, String&gt; hmap; // violation
+ *     Queue&lt;Integer&gt; intqueue; // OK
+ *
+ *     java.lang.IllegalArgumentException illegalex; // OK
+ *     java.util.TreeSet treeset; // violation
+ *   }
+ *
+ * }
+ * </pre>
+ * <p>
+ * To configure the Check so that particular tokens are checked:
+ * </p>
+ * <pre>
+ * &lt;module name="IllegalType"&gt;
+ *   &lt;property name="tokens" value="METHOD_DEF"/&gt;
+ * &lt;/module&gt;
+ * </pre>
+ * <pre>
+ * public class Test extends TreeSet { // OK
+ *   public &lt;T extends java.util.HashSet&gt; void method() { // violation
+ *     LinkedHashMap&lt;Integer, String&gt; lhmap =
+ *     new LinkedHashMap&lt;Integer, String&gt;(); // OK
+ *
+ *     java.lang.IllegalArgumentException illegalex; // OK
+ *     java.util.TreeSet treeset; // Ok
+ *   }
+ *
+ *   public &lt;T extends java.util.HashSet&gt; void typeParam(T) {} // violation
+ *
+ *   public void fullName(TreeSet a) {} // OK
+ *
+ * }
+ * </pre>
+ * <p>
+ * To configure the Check so that it ignores function() methods:
  * </p>
  * <pre>
  * &lt;module name=&quot;IllegalType&quot;&gt;
- *   &lt;property name=&quot;ignoredMethodNames&quot; value=&quot;getInstance&quot;/&gt;
+ *   &lt;property name=&quot;ignoredMethodNames&quot; value=&quot;function1&quot;/&gt;
```

Comment:
please change ignored method name to `function` as description says

Reply:
done.